### PR TITLE
[Table/create/240325]  Create Migrations for some tables

### PR DESCRIPTION
**Pull request by kaoribundo**
[Table/Create/240325] -Create Migration Tables
This is the branch based on the branch which Chie-san created below.
https://github.com/kredo-school/18tha-gj_mall/tree/Table/Create/240323

Chie-sann added Tables names and columns.
- Customers
- Sellers
- Admins
- Ads
- Reviews
- Favorite
- Inquiry_genres
- inquiry_status
- inquires
- order_lines
- order_status
- shipping_method
- shop_orders
- shopping_card_items
- product_status
- categories
- product_image
- product_images
- product_details

In this branch I added foreign_keys and modify the datatype of
- product_details.weight from unsignedbiginteger to text
- the contents are like 100g , 30kg , 20mg×12 ...

### CODE REVIEW

File: database/migrations/2024_03_23_064740_create_ads_table.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('ads', function (Blueprint $table) {
+            $table->id();
+            $table->string('title');
+            $table->text('content');
+            $table->longText('cover');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_064740_create_ads_table.php]: what's "cover" ? Or you could use other term for this column

File: database/migrations/2024_03_23_065951_create_reviews_table.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('reviews', function (Blueprint $table) {
+            $table->id();
+            $table->string('title');
+            $table->text('content');
+            $table->unsignedBigInteger('rating')->nullable()->comment('min:0, max:5');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_065951_create_reviews_table.php]: if we only expect 0 - 5 values for this column, use tiny int

File: database/migrations/2024_03_23_071253_create_inquiry_status_table.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('inquiry_status', function (Blueprint $table) {
+            $table->id();
+            $table->string('status')->unique()->comment('1:unsolved, 2:answer, 3:solved');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_071253_create_inquiry_status_table.php]: use int for status

File: database/migrations/2024_03_23_072225_create_order_lines_table.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('order_lines', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('product_id');
+            $table->unsignedBigInteger('order_id');
+            $table->unsignedBigInteger('qty');
+            $table->unsignedBigInteger('price');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_072225_create_order_lines_table.php]: use double for price

File: database/migrations/2024_03_23_072523_create_order_status_table.php [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('order_status', function (Blueprint $table) {
+            $table->id();
+            $table->string('status')->unique()->comment('1:Waiting for packing -> 2:Waiting for acceptance -> (6:Cancellation -> 7:Refund completed ->) 3:Acceptance complete -> 4:In transit -> 5:Delivery completed');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_072523_create_order_status_table.php]: use int for status

File: database/migrations/2024_03_23_072912_create_shipping_method_table.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('shipping_method', function (Blueprint $table) {
+            $table->id();
+            $table->string('name')->unique()->nullable();
+            $table->unsignedBigInteger('price');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_072912_create_shipping_method_table.php]: use double for price

File: database/migrations/2024_03_23_073129_create_shop_orders_table.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    public function up(): void
+    {
+        Schema::create('shop_orders', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('customer_id');
+            $table->unsignedBigInteger('address_id');
+            $table->unsignedBigInteger('shipping_method_id');
+            $table->unsignedBigInteger('status_id');
+            $table->unsignedBigInteger('order_total');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_073129_create_shop_orders_table.php]: is this the total amount or the total quantity ? if amount, please use double

File: database/migrations/2024_03_23_073625_create_product_status_table.php [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('product_status', function (Blueprint $table) {
+            $table->id();
+            $table->string('status')->unique()->comment('1:Exhibit request -> 2:Waiting for valuation -> 3:Evaluation -> (7:Suspended ->) 4:Waiting for display(Coming Soon) -> 5:Exhibited -> 6:Sold out');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_073625_create_product_status_table.php]: use int for status

File: database/migrations/2024_03_23_074226_create_product_images_table.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('product_images', function (Blueprint $table) {
+            $table->id();
+            $table->longText('image');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_074226_create_product_images_table.php]: no need for long text, use string

File: database/migrations/2024_03_23_074510_create_product_details_table.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('product_details', function (Blueprint $table) {
+            $table->id();
+            $table->text('size');
```
Comment (Ninit29) [inline database/migrations/2024_03_23_074510_create_product_details_table.php:16]: use double for size and weight

File: database/migrations/2024_03_25_105724_create_customers_table.php [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    public function up(): void
+    {
+        Schema::create('customers', function (Blueprint $table) {
+            $table->id();
+            $table->string('first_name');
+            $table->string('last_name');
+            $table->string('email')->unique();
+            $table->string('phone_number')->nullable()->unique();
+            $table->string('password');
+            $table->longText('avatar')->nullable();
```
Comment (Ninit29) [inline database/migrations/2024_03_25_105724_create_customers_table.php]: use string for avatar

File: database/migrations/2024_03_25_111021_create_sellers_table.php [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    public function up(): void
+    {
+        Schema::create('sellers', function (Blueprint $table) {
+            $table->id();
+            $table->string('first_name');
+            $table->string('last_name');
+            $table->string('email')->unique();
+            $table->string('phone_number')->nullable()->unique();
+            $table->string('password');
+            $table->longText('avatar')->nullable();
```
Comment (Ninit29) [inline database/migrations/2024_03_25_111021_create_sellers_table.php]: use string for avatar

File: database/migrations/2024_03_25_111409_create_admin_table.php [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    public function up(): void
+    {
+        Schema::create('admin', function (Blueprint $table) {
+            $table->id();
+            $table->string('first_name');
+            $table->string('last_name');
+            $table->string('email')->unique();
+            $table->string('phone_number')->nullable()->unique();
+            $table->string('password');
+            $table->unsignedBigInteger('role')->default(2)->comment('1:admin, 2:stuff, 3:translater, 4:assessor, 5:delivery');
```
Comment (Ninit29) [inline database/migrations/2024_03_25_111409_create_admin_table.php]: use int not big int

File: database/migrations/2024_03_23_064740_create_ads_table.php (hunk 1 above)
Comment (kaoribundo) [reply database/migrations/2024_03_23_064740_create_ads_table.php]: We will use this column for images.
I will change this to "image_name".

We are going to use these images for HomePage cover slides, so we meant this title.

File: database/migrations/2024_03_23_071253_create_inquiry_status_table.php (hunk 3 above)
Comment (kaoribundo) [reply database/migrations/2024_03_23_071253_create_inquiry_status_table.php:16]: We use id for primarykey and the content of id in status, so we think status should be string , but rather fix to 
`string('status' , 15)`

File: database/migrations/2024_03_23_073129_create_shop_orders_table.php (hunk 7 above)
Comment (kaoribundo) [reply database/migrations/2024_03_23_073129_create_shop_orders_table.php]: Yes it is the total price amount . We will use double.

File: database/migrations/2024_03_23_074510_create_product_details_table.php (hunk 10 above)
Comment (kaoribundo) [reply database/migrations/2024_03_23_074510_create_product_details_table.php:16]: We want to include the units of size such as L:30cm W:15cm H:10cm.
For the weight the same reasons(ex.300g, 20kg). Seller can edit the size and weight in any style.

File: database/migrations/2024_03_23_071253_create_inquiry_status_table.php (hunk 3 above)
Comment (kaoribundo) [reply database/migrations/2024_03_23_071253_create_inquiry_status_table.php]: Sorry this is a kind of master table , so this one must be not changable. I will modify this again.

File: database/migrations/2024_03_23_072523_create_order_status_table.php (hunk 5 above)
Comment (kaoribundo) [reply database/migrations/2024_03_23_072523_create_order_status_table.php]: Sorry this is a kind of master table , so this one must be not changable. I will modify this again.

File: database/migrations/2024_03_23_073625_create_product_status_table.php (hunk 8 above)
Comment (kaoribundo) [reply database/migrations/2024_03_23_073625_create_product_status_table.php]: Sorry this is a kind of master table , so this one must be not changable. I will modify this again.

**Review by KobeSergio — APPROVED** [review]

**Pull request merged**